### DISCUSSION TITLE
Multiple Database Support and 2.1.0

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#1258] [General]
With 2.0.2 being released today, it is time to focus on 2.1.0 where the primary focus is Multiple Database Support ( originally described in #964 ). I will be merging PR #1239 tonight which Charles Nurse submitted last week into the Dev branch. Please note that this PR contains extensive changes to the core framework and is expected to require some significant refactoring and testing to get it 100% ready for release. As a result it would be very unwise to perform a new installation or upgrade any prior installation to the latest Dev branch prior to the 2.1.0 release. At this time the Dev branch should be reserved for temporary development environments where you are comfortable with deleting your database and starting over with a clean install.

**Comment by sbwalker** [comment]
Please note that performing an installation of the Dev branch when running on WebAssembly is not currently supported at the moment ( this is one of the areas where refactoring is required ).

**Comment by hishamco** [comment]
@sbwalker I will mention the CI/CD again, could you please make this done, we need to write extensive tests to make sure nothing break

**Comment by sbwalker** [comment]
Please note that although the Dev branch builds fine, if you attempt a default installation on the Server runtime at this moment it will not work due to some integration issues when merging #1239. This should be resolved today.

**Reply by hishamco** [reply]
I will have a try new changes in the dev branch once I finish a small localization related PRs

**Reply by hishamco** [reply]
> lease note that although the Dev branch builds fine, if you attempt a default installation on the Server runtime at this moment it will not work due to some integration issues when merging #1239. This should be resolved today.

Now it's working fine, but I notice some latency during the installation process

**Reply by sbwalker** [reply]
#1262 resolved the installation issues when running on Server runtime - WebAssembly still needs more work.

**Reply by hishamco** [reply]
One more thing I didn't see the latest changes I made in the `dev` does all the commits for `2.0.2` are there?

**Reply by sbwalker** [reply]
@hishamco all of the 2.0.2 changes should be in dev... unless there were mistakenly removed when @cnurse resolved conflicts for his PR. Are there specific changes you cannot find?

**Reply by hishamco** [reply]
I will have a look to the commits logs for the `dev`

**Comment by hishamco** [comment]
Few notes after a quick look to the new changes:

- Make SQLite a default database while it's file-based and zero configuration
- Move SQL Server into a new project
- Organize the database projects into `Providers/Database` folder
- Create packages for the database projects to make it easier for distribution

**Reply by sbwalker** [reply]
@hishamco please hold off on submitting any PRs related to the Multi-Database enhancements until I have a chance to discuss with @cnurse

**Reply by hishamco** [reply]
Ok, I just push one PR that's not hurt just a refactoring, I listed some of the intial notes above, I will continue on them once you resolve the integration issues

**Reply by hishamco** [reply]
BTW can @cnurse discuss with the community all the related things about multi database support or he is busy?

**Comment by sbwalker** [comment]
Multiple Database Support has been evolving and stabilizing over the past month, and I am very pleased to say that we are getting very close to having a solution which meets all of the goals described in #964. The Oqtane 2.1 release will contain support for SQL Server & LocalDB, MySQL, SQLite, and PostgreSQL - which will provide native database support for all of the most popular databases on every platform. The architecture will also allow additional database providers to be developed and integrated with the framework. I would like to thank @pcancela for his MySQL POC and @cnurse for all of his efforts in getting this across the finish line!

**Comment by sbwalker** [comment]
The 2.1 release was finally published on GitHub today... I will be writing the official marketing announcement this weekend. This release contains some game changing enhancements... a huge thank you to the community for the many  contributions!!

**Comment by sbwalker** [comment]
Public announcement for 2.1 release published here: https://www.oqtane.org/Resources/Blog/PostId/548/oqtane-21-now-supports-multiple-databases
